src/uct/api/uct.h

```diff
@@ -396,6 +396,17 @@ typedef enum {
 } uct_memory_type_t;
 
 
+/*
+ * @ingroup UCT_MD
+ * @brief  Memory type owned status
+ */
+typedef enum {
+    UCT_MD_MEM_TYPE_OWNED = 0,          /**< MD owns the memory type */
+    UCT_MD_MEM_TYPE_NOT_OWNED,          /**< MD not owns the memort type */
+    UCT_MD_MEM_TYPE_NOT_DETECTABLE      /**< MD can't detect memory owned or not */
```

Comment:
why do we need that?

Reply:
it seems semantically it is not clean to return NOT_OWNED for the HOST memory  on  non-cuda MDs.

Reply:
an MD should return either "true" (definitely owned) or "false" (definitely not owned). "false" does not necessarily mean it's host memory; however, if all memory domains returned "not owned" UCP would conclude this is host memory. This conclusion would also be correct assuming we have support in UCX for all possible memory types in the application.

Reply:
we can return "true" or "false" if we conclude that it is host memory if all MD return "no owned"
will update PR

Reply:
BTW, when i said "true"/"false" i didn't mean that we need to define such constants, just plain old 0/1

Reply:
updated PR.